CovertInfiltration/Src/CovertInfiltration/Classes/XComGameState_ActivityChain.uc

```diff
@@ -218,6 +255,27 @@ function CurrentStageHasCompleted (XComGameState NewGameState)
 		{
 			m_Template.CleanupChain(NewGameState, self);
 		}
+		
+		// Fire off chain complications
+		if (m_Template.bAllowComplications)
+		{
+			TemplateManager = class'X2StrategyElementTemplateManager'.static.GetStrategyElementTemplateManager();
+		
+			for (i = 0; i < Complications.Length; i++)
+			{
+				ChainComp = X2ComplicationTemplate(TemplateManager.FindStrategyElementTemplate(Complications[i]));
+
+				CompRoll = `SYNC_RAND_STATIC(100);
+			
+				if (CompRoll < CompChances[i])
+				{
+					if (EndReason == eACER_Complete)
+						ChainComp.OnChainComplete(NewGameState, self);
+					if (EndReason == eACER_ProgressBlocked)
+						ChainComp.OnChainBlocked(NewGameState, self);
```

Comment:
Check for `ChainComp.OnChainBlocked != none` (same for the other one) to prevent warnings in the log